pokemongo_bot/cell_workers/evolve_pokemon.py

```diff
@@ -65,19 +70,38 @@ def work(self):
                 candy = inventory.candies().get(pokemon.pokemon_id)
                 pokemon_to_be_evolved = pokemon_to_be_evolved + min(candy.quantity / (pokemon.evolution_cost - 1), filtered_dict[pokemon.pokemon_id])
 
-        if pokemon_to_be_evolved >= self.min_pokemon_to_be_evolved:
+        self._print_check(pokemon_to_be_evolved)
+
+        has_minimum_to_evolve = pokemon_to_be_evolved >= self.min_pokemon_to_be_evolved
+        if has_minimum_to_evolve:
             if self.use_lucky_egg:
                 self._use_lucky_egg()
             cache = {}
             for pokemon in filtered_list:
                 if pokemon.can_evolve_now():
                     self._execute_pokemon_evolve(pokemon, cache)
 
+    def _print_check(self, pokemon_to_be_evolved):
+        has_minimum_to_evolve = pokemon_to_be_evolved >= self.min_pokemon_to_be_evolved
+        result_message = ("Gotta catch`em all!", "Gotta evolv`em all!")[has_minimum_to_evolve]
+        self.emit_event(
+            'pokemon_evolve_check',
+            formatted='Checking... Has {has}, needs {needs}. {message}',
+            data={
+                'has': pokemon_to_be_evolved,
+                'needs': self.min_pokemon_to_be_evolved,
+                'message': result_message
+            }
+        )
+
     def _should_run(self):
         if not self.evolve_list or self.evolve_list[0] == 'none':
             return False
-        return True
-    
+        return self.next_update is None or datetime.now() >= self.next_update
```

Comment:
Hmmm yes I see what you mean and I agree with you, but I'd rather not change the behavior of the task since the purpose of this PR is for giving a status update, and not to reduce frequency of evolve task. 

Assuming that **not** all users read the documentation, we are essentially giving users the power/ability to screw up the evolve task.

IMO, if the frequency of evolve task should be reduced, it should be in another PR. What do you think?